native/src/utils/renderHtml.ts

```diff
@@ -71,34 +84,144 @@ const renderJS = (cacheDictionary: Record<string, string>) => `
-    window.addEventListener('resize', adjustHeight);
-    var details = document.querySelectorAll("details")
-    details.forEach(detail => detail.addEventListener("toggle", adjustHeight))
+    window.addEventListener('load', adjustHeight)
+    window.addEventListener('resize', adjustHeight)
+    const details = document.querySelectorAll('details')
+    details.forEach(detail => detail.addEventListener('toggle', adjustHeight))
   })();
+
+  (function handleIframes() {
+    function capitalizeFirstLetter(word) {
+      return word.charAt(0).toUpperCase() + word.slice(1)
+    }
+
+    function showMessage(text, element, className) {
+      const textNode = document.createTextNode(text)
+      if (className) {
+        element.classList.add(className)
+      }
+      element.appendChild(textNode)
+    }
+
+    function showSettingsButton(element) {
+      function onClickHandler () {
+        window.ReactNativeWebView.postMessage(JSON.stringify({ type: '${SETTINGS_MESSAGE_TYPE}', openSettings: true }))
+      }
+
+      const buttonLabel = ${JSON.stringify(t('layout:settings'))}
+      const button = document.createElement('button')
+      button.name = 'opt-in-settings-button'
+      button.innerHTML = buttonLabel
+      button.id = 'opt-in-settings-button'
+      button.onclick = onClickHandler
+      element.appendChild(button)
+    }
+
+    function showMessageWithSettings(text, iframeContainer) {
+      showMessage(text, iframeContainer, 'iframe-info-text')
+      showSettingsButton(iframeContainer)
+
+    }
+
+    function showOptIn(text, iframeContainer, source) {
+      function onClickHandler () {
+        window.ReactNativeWebView.postMessage(JSON.stringify({ type: '${IFRAME_MESSAGE_TYPE}', allowedSource: { type: source, allowed: true } }))
+      }
+
+      const checkbox = document.createElement('input')
+      checkbox.type = 'checkbox'
+      checkbox.name = 'opt-in-checkbox'
+      checkbox.id = 'opt-in-checkbox'
+      checkbox.onclick = onClickHandler
+      const label = document.createElement('label')
+      label.htmlFor = 'opt-in-checkbox'
+      label.appendChild(document.createTextNode(text))
+      iframeContainer.appendChild(label)
+      iframeContainer.appendChild(checkbox)
+    }
+
+    function showBlockMessageWithSettings(text, iframeContainer) {
+      showMessage(text, iframeContainer)
+      showSettingsButton(iframeContainer)
+    }
+
+    function handleAllowedIframeSources(iframe, allowedExternalSourcePermissions, iframeSource, iframeContainer) {
+      if (externalSourcePermissions.some(source => source.type === iframeSource && source.allowed)) {
+        // Add do not track parameter (only working for vimeo)
+        if (iframeSource === 'Vimeo') {
+          const url = new URL(iframe.src)
+          url.searchParams.append('dnt', '1')
+          iframe.setAttribute('src', url.href)
+        }
+        const message = ${JSON.stringify(
+          t('remoteContent:knownResourceContentMessageOne'),
```

Comment:
i couldn't find a proper way to pass variables (except of defined strings) to a passed function inside this `JSON.stringify`
If someone finds a solution that would simplify a lot.

Reply:
I think its just not possible sadly since the functions are evaluated before the html is executed.

One idea though: I think we should write the messages in the way that they are universal and not iframe source specific, e.g.
```
vimeo.com
External content from this source was blocked. You can allow this source in the settings.
```
Then it should a) be easier to read/implement and b) more error proof in other translations (bc of things like word order etc.).